src/Common/DateLUTImpl.h

```diff
@@ -381,19 +381,19 @@ class DateLUTImpl
     {
         const LUTIndex i = toLUTIndex(v);
         if constexpr (std::is_unsigned_v<DateOrTime> || std::is_same_v<DateOrTime, DayNum>)
-            return lut_saturated[i - lut[i].day_of_month + lut[i].days_in_month].date;
+            return lut_saturated[i + (lut[i].days_in_month - lut[i].day_of_month)].date;
```

Comment:
You could drop the parentheses:

```cpp
i + lut[i].days_in_month - lut[i].day_of_month
```

Same in l. 386, 394, 396.

Reply:
Dropping parentheses will call 2 "expensive" DateLUT operators:
first time `operator +` and then `operator - `
Agree?
While keeping the parentheses will call only one DateLUT `operator +`

Reply:
`i` is a StrongTypedef (base/base/strong_typedef.h) which is basically an `UInt32`.
`lut[i].days_in_month` is a `UInt8`.
`lut[i].day_of_month` is a `UInt8`.

`i + (lut[i].days_in_month - lut[i].day_of_month)` means
1. Read `lut[i]` and then it's member `days_in_month`.
2. Read `lut[i]` and then it's member `day_of_month`.
3. Subtract two `UInt8`s.
4. Add a `UInt32` and a `UInt8`.

`i + lut[i].days_in_month - lut[i].day_of_month` means
1. Read `lut[i]` and then it's member `days_in_month`.
2. Add a `UInt32` and a `UInt8`.
3. Read `lut[i]` and then it's member `day_of_month`.
4. Subtract a `UInt8`s from an `UInt32`.

Maybe I really miss something (it's sort of late 😉) but I count the same amount of work done.